apps/journeys-admin/src/components/Editor/Slider/Settings/CanvasDetails/Footer/Menu/MenuActionButton/useMenuBlockCreateMutation/useMenuBlockCreateMutation.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  ApolloCache,
+  MutationHookOptions,
+  MutationTuple,
+  gql,
+  useMutation
+} from '@apollo/client'
+
+import { CARD_FIELDS } from '@core/journeys/ui/Card/cardFields'
+import { useJourney } from '@core/journeys/ui/JourneyProvider'
+import { STEP_FIELDS } from '@core/journeys/ui/Step/stepFields'
+import { TYPOGRAPHY_FIELDS } from '@core/journeys/ui/Typography/typographyFields'
+
+import {
+  MenuBlockCreate,
+  MenuBlockCreateVariables
+} from '../../../../../../../../../../__generated__/MenuBlockCreate'
+
+export const MENU_BLOCK_CREATE = gql`
+  ${STEP_FIELDS}
+  ${CARD_FIELDS}
+  ${TYPOGRAPHY_FIELDS}
+  mutation MenuBlockCreate(
+    $journeyId: ID!
+    $stepBlockCreateInput: StepBlockCreateInput!
+    $cardBlockCreateInput: CardBlockCreateInput!
+    $typographyBlockCreateInput: TypographyBlockCreateInput!
+    $journeyUpdateInput: JourneyUpdateInput!
+  ) {
+    step: stepBlockCreate(input: $stepBlockCreateInput) {
+      ...StepFields
+      x
+      y
+    }
+    card: cardBlockCreate(input: $cardBlockCreateInput) {
+      ...CardFields
+    }
+    typography: typographyBlockCreate(input: $typographyBlockCreateInput) {
+      ...TypographyFields
+    }
+    journeyUpdate(id: $journeyId, input: $journeyUpdateInput) {
+      menuStepBlock {
+        ...StepFields
+      }
```

Comment:
This should fix your cache problem. Returning the id is how the cache identifies which record to update. Can you also add this to your other hooks?
```suggestion
    journeyUpdate(id: $journeyId, input: $journeyUpdateInput) {
      id
      menuStepBlock {
        ...StepFields
      }
```